File: 프로그래머스/2/49994. 방문 길이/방문 길이.py
def move(now_x, now_y, dir):
    if dir == 'U':
        nx, ny = now_x, now_y + 1
    elif dir == 'L':
        nx, ny = now_x -1, now_y
    elif dir == 'R':
        nx, ny = now_x + 1, now_y
    else:
        nx, ny = now_x, now_y-1
    return (nx, ny)

def solution(dirs):
    ans = set()
    x, y = 0, 0 

    for d in dirs:
        nx, ny = move(x,y,d)
        if -5<=nx<=5 and -5<=ny<=5:
            ans.add(((x,y), (nx, ny)))
            ans.add(((nx,ny),(x,y)))
            x, y = nx, ny


    return len(ans)//2

File: 프로그래머스/2/49994. 방문 길이/방문 길이.js
const solution = (dirs) => {
  const move = (x, y, dir) => {
    if (dir == "U") {
      [nx, ny] = [x, y + 1];
    } else if (dir == "L") {
      [nx, ny] = [x - 1, y];
    } else if (dir == "R") {
      [nx, ny] = [x + 1, y];
    } else {
      [nx, ny] = [x, y - 1];
    }
    return [nx, ny];
  };

  let x = 0,
    y = 0;
  const ans = new Set();

  for (let d of dirs) {
    const [nx, ny] = move(x, y, d);
    if (nx >= -5 && nx <= 5 && ny >= -5 && ny <= 5) {
      ans.add(`${x}${y}${nx}${ny}`);
      ans.add(`${nx}${ny}${x}${y}`);
      [x, y] = [nx, ny];
    }
  }

  return ans.size / 2;
};
